### ISSUE TITLE
Issue: Typed_params breaks on ruby 3.3.0

### ISSUE THREAD

**Issue by dylandamsma** [keygen-sh/typed_params#6]
First of all, fantastic work on this gem. It's precisely what I was looking for to move fast in a very small rails + mobile team, without having to go back and forth on "What should the value be here?".

However, unfortunately found that the gem breaks on Ruby 3.3.0

```
../.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/typed_params-1.1.0/lib/typed_params/mapper.rb:58: anonymous block parameter is also used within block (SyntaxError)
../.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/typed_params-1.1.0/lib/typed_params/mapper.rb:60: anonymous block parameter is also used within block
```

Please see this change: https://rubyreferences.github.io/rubychanges/3.3.html#anonymous-parameters-forwarding-inside-blocks-are-disallowed

Unfortunately, I'm not so well versed myself in the problem/solution here, otherwise I'd have opened a PR, apologies 🙏

**Comment by ezekg** [comment]
This looks like [a Ruby bug](https://bugs.ruby-lang.org/issues/20090), since the blocks aren't actually defining anonymous blocks, but rather using [the anonymous block defined in the top level method](https://github.com/keygen-sh/typed_params/blob/master/lib/typed_params/mapper.rb#L50). but I'll fix here in the meantime since it's not a big deal. All we have to do is rename the `&` parameter to `&block`. I'm out of town, but will get a new release cut on or around the 1st.

**Comment by ezekg** [comment]
Fixed in v1.1.1.

**Comment by dylandamsma** [comment]
Really appreciate the reply and the fix as well as the explanation. I'll do my best to understand more!
